contracts/Hub.sol

```diff
@@ -0,0 +1,138 @@
+        for(uint i = 0; i < authorizedAddresses.length; i++) {
+            if(authorizedAddresses[i] == msg.sender) {
+                isAuthorized = true;
+            }
+        }
+        require(isAuthorized);
+        _;
+    }
+
+    constructor() {
+    }
+
+    /**
+     * @notice Add an address to the list of authorized addresses
+     * @param addr address that gets added 
+     */
+    function addAuthorizedAddress(address addr) public onlyOwner{
+        authorizedAddresses.push(addr);
+    }
+
+    /**
+     * @notice updates the address asssociated with an synthAsset
+     * @param tickerSymbol identifier of synthAsset
+     * @param addr new contract address of synthAsset 
+     */
+    function updateContractProxy(string memory tickerSymbol, address addr) public onlyAuthorizedAddresses{
+        tickersymbolToAddress[tickerSymbol] = addr; 
+    }
+
+    /**
+     * @notice mints synthAssets to a specific address
+     * @param receiver address to which token gets minted
+     * @param amount amount of token that gets minted
+     * @param tickerSymbol identifier of which token gets minted
+     */
+    function mintToken(address receiver, uint256 amount, string memory tickerSymbol) public onlyAuthorizedAddresses{
+        //TODO: mintToken for a specific synthAsset. 
+        //How do we interact with contracts we don't have the interface for
+        //-> every synthAsset contract
+        //maybe heritage??
+        minterContract.mint(receiver, amount); 
+    }
+
+    /**
+     * @notice sets the Minter contract address
+     * @param _minterAddress address of Minter contract
+     */
+    function setMinterContract(address _minterAddress) public onlyOwner{
+        minterContract = Minter(_minterAddress);
+    }
+
+    /**
+     * @notice sets the Fund contract address
+     * @param _fundAddress address of Fund contract
+     */
+    function setFundContract(address _fundAddress) public onlyOwner{
+        fundContract = Fund(_fundAddress);
+    }
+
+    /**
+     * @notice sets the Storage contract address
+     * @param _storageAddress address of Storage contract
+     */
+    function setStorageContract(address _storageAddress) public onlyOwner{
+        storageContract = Storage(_storageAddress);
+    }
+
+    /**
+     * @notice checks the collateral ratio of an address.
+     * @dev for all assets combined. No individual / isolated positions for now.
+     * @dev price checking of colllateral too -> volatile collateral / depegging of stable collateral
+     * @param addr address of user whom collateral ratio is to be checked.
+     * @param collateralTickerSymbol identifier of token used for collateral#
+     * @return collateral ratio
+     */
+    function getCollateralRatio(address addr, string memory collateralTickerSymbol) public view returns (uint256) {
+         
+         //Check amount funded
+         uint256 amountFunded = fundContract.getAmountFundedByAddress(msg.sender);
```

Comment:
we still need to implement this function yet, but let's call it `getCollateralByAddress` again because it declares the purpose of the variable better